core/base/src/TSystem.cxx

```diff
@@ -2817,6 +2817,13 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
                           const char *build_dir,
                           UInt_t dirmode)
 {
+   // The function accesses the filesystem and the interpreter in various places
+   // thus needs locking. Possible situations that may trigger data races
+   // include:
+   // - Multiple threads compiling different macros
+   // - One thread calling this functions while others are accessing the
+   //   interpreter for various reasons
+   R__LOCKGUARD(gInterpreterMutex);
```

Comment:
Is this the minimum size of the critical section, i.e. can the lockguad be instantiated after this point preserving thread safety?

Reply:
It is not.  Scanning the code, it looks like the potential race conditions starts around old line 2999 (however the race on line 2999 would also be a race with other processes) or around old line 3020.